newIDE/app/src/SceneEditor/ObjectsAdditionalService.js

```diff
@@ -0,0 +1,118 @@
+// @flow
+import { type AlertMessageIdentifier } from '../MainFrame/Preferences/PreferencesContext';
+
+/*
+ * Define additional logic which executes after an object/instance has been created.
+ * Also, InfoBar can be used which could notify users of the additional changes.
+ * Declare new identifier for infoBar in Mainframe/Preferences/PreferenceContext
+ * and add it in hints/explaination list.
+ */
+
+export type InfoBarDetails = {
+  identifier: AlertMessageIdentifier,
+  message: string,
+  touchScreenMessage: string,
+  show: boolean,
+};
+
+type InfoBarEvent = 'onObjectAdded' | 'onInstanceAdded';
+
+type AdditionalServices = {
+  onObjectAdded: (
+    object: gdObject,
+    project: gdProject,
+    layout: gdLayout
+  ) => void,
+  onInstanceAdded: (
+    instance: gdInitialInstance,
+    project: gdProject,
+    layout: gdLayout
+  ) => void,
+  getInfoBarDetails: (infoBarEvent: InfoBarEvent) => InfoBarDetails,
+};
+
+export default {
+  getServices(
+    objectOrInstance: gdObject | gdInitialInstance,
+    layout: gdLayout,
+    isInstance: boolean
+  ): ?AdditionalServices {
+    let objectTypeServices = null;
+    if (isInstance) {
+      objectTypeServices = this.objectType[
+        // $FlowFixMe
```

Comment:
@4ian how should I fix this? I'm not sure if I could really check `objectOrInstance` is `gdObject` or `gdInitialInstance`, is there a way of doing that?

Reply:
I think you're trying to fit two different stuff in a single function :)
We don't really need something to return these "additional service" because we just care about either running `onObjectAdded` or `onInstanceAdded `

Instead, I would just export these two functions and use them directly (passing them the object or instance, layout, project, and maybe a callback to set the infobar or whatever is needed).